mteb/models/msclap_models.py

```diff
@@ -0,0 +1,278 @@
+                sr = audio_item.get("sampling_rate", self.sampling_rate)
+
+                if isinstance(audio, np.ndarray):
+                    audio = torch.from_numpy(audio).float()
+                elif isinstance(audio, list):
+                    audio = torch.tensor(audio, dtype=torch.float32)
+                else:
+                    audio = audio.float()
+
+                # Resample if needed
+                if sr != self.sampling_rate:
+                    resampler = torchaudio.transforms.Resample(sr, self.sampling_rate)
+                    audio = resampler(audio)
+
+                return audio.squeeze()
+
+            elif "path" in audio_item:
+                return self._load_audio_file(audio_item["path"])
+
+        elif isinstance(audio_item, (np.ndarray, torch.Tensor)):
+            if isinstance(audio_item, np.ndarray):
+                audio_item = torch.from_numpy(audio_item)
+            return audio_item.squeeze().float()
+
+        elif isinstance(audio_item, str):
+            return self._load_audio_file(audio_item)
+
+        else:
+            raise ValueError(f"Unsupported audio format: {type(audio_item)}")
+
+    def _load_audio_file(self, path: str) -> torch.Tensor:
+        """Load audio file and convert to proper format"""
+        waveform, sample_rate = torchaudio.load(path)
+        waveform = waveform.float()
+
+        if sample_rate != self.sampling_rate:
+            resampler = torchaudio.transforms.Resample(sample_rate, self.sampling_rate)
+            waveform = resampler(waveform)
+
+        return waveform.squeeze()
+
+    def get_audio_embeddings(
+        self,
+        audio: AudioBatch,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        """Get audio embeddings using direct tensor processing (no temp files)"""
+        all_features = []
+
+        if isinstance(audio, DataLoader):
+            # Process all batches
+            for batch in tqdm(audio, desc="Processing audio batches"):
+                batch_features = self._process_audio_batch(batch)
+                all_features.extend(batch_features)
+        else:
+            # Process single batch
+            batch_features = self._process_audio_batch(audio)
+            all_features.extend(batch_features)
+
+        return np.vstack(all_features)
+
+    def _process_audio_batch(self, batch) -> list[np.ndarray]:
+        """Process a batch of audio items and return embeddings"""
+        batch_features = []
+
+        for item in batch:
+            # Convert to tensor
+            audio_tensor = self._process_audio_to_tensor(item)
+
+            # Ensure it's in the right format [batch_size, samples]
+            if audio_tensor.dim() == 1:
+                audio_tensor = audio_tensor.unsqueeze(0)  # Add batch dimension
+
+            audio_tensor = audio_tensor.to(self.device)
+            # Get embeddings using the internal audio encoder
+            with torch.no_grad():
+                # Use the internal method
```

Comment:
Might want to reference the comment here as well: https://github.com/microsoft/CLAP/blob/e8a6467b87cd85716e20c6a008126150d9740be0/msclap/CLAPWrapper.py#L295
> [0] the audio emebdding, [1] has output class probabilities